### PR TITLE
Allow addressspace spec changes (#3541)

### PR DESCRIPTION
**Pull request by lulf**
Same as #3541

### CODE REVIEW

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

**Comment by lulf** [comment]
@k-wall The resolved authservice settings are now part of the config.

**Comment by lulf** [comment]
@enmasse-ci run tests profile=smoke

**Comment by lulf** [comment]
@enmasse-ci run tests profile=upgrade

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

**Comment by lulf** [comment]
@enmasse-ci run tests profile=shared

**Comment by kornys** [comment]
@lulf Ive added one quick test, see results please, seems in standard addressspace persistent messages are not present after auth service switch.

**Comment by kornys** [comment]
@enmasse-ci run tests testcase=AuthServiceTest#testSwitchAuthService

File: address-space-controller/src/main/java/io/enmasse/controller/CreateController.java [hunk 1]
```diff
@@ -163,47 +168,46 @@ public AddressSpace reconcileActive(AddressSpace addressSpace) throws Exception
...
                 kubernetes.apply(resourceList,desiredInfraConfig.getUpdatePersistentVolumeClaim());
                 eventLogger.log(AddressSpaceUpgraded, "Upgraded address space", Normal, ControllerKind.AddressSpace, addressSpace.getMetadata().getName());
                 InfraConfigs.setCurrentInfraConfig(addressSpace, desiredInfraConfig);
-                addressSpace.putAnnotation(AnnotationKeys.APPLIED_PLAN, addressSpace.getSpec().getPlan());
+                AppliedConfig.setCurrentAppliedConfig(addressSpace, desiredConfig);
             } else {
                 log.info("Version of desired config ({}) does not match controller version ({}), skipping upgrade", desiredInfraConfig.getVersion(), version);
             }
-        } else if (!addressSpace.getSpec().getPlan().equals(appliedPlan)) {
+        } else if (!desiredConfig.equals(appliedConfig)) {
```
Comment (k-wall) [inline address-space-controller/src/main/java/io/enmasse/controller/CreateController.java]: The code paths for unequal infra configs and unequal applied configs now look similar.  The only material difference is the infraconfig version check - which I'm not sure remains important.  It makes me wonder if the `AppliedConfig` object should carry the infraconfig object too, allowing the infra config annotation and the `else if` to be removed.

Comment (lulf) [reply address-space-controller/src/main/java/io/enmasse/controller/CreateController.java]: I agree they should be merged. However, I would perhaps like to do this separately to isolate the effects.

**Comment by obabec** [comment]
@enmasse-ci run tests testcase=ApiServerTest#testChangeAddressSpaceType

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ApiServerTest#testChangeAddressSpaceType

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ApiServerTest#testChangeAddressSpaceType

**Comment by lulf** [comment]
@enmasse-ci run tests

**Comment by kornys** [comment]
@enmasse-ci run tests testcase=AuthServiceTest#testSwitchAuthService

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=AuthServiceTest#testSwitchAuthService

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=AuthServiceTest#testSwitchAuthService

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=InfraTestBrokered

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=infra.**

**Comment by lulf** [comment]
@enmasse-ci run tests

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=io.enmasse.systemtest.isolated.plans.standard.PlansTestStandard#testCannotReplaceAddressSpacePlanStandard

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=io.enmasse.systemtest.isolated.plans.standard.PlansTestStandard#testCannotReplaceAddressSpacePlanStandard

**Pull request merged**